src/WebJobs.Script.WebHost/Metrics/LinuxContainerMetricsPublisher.cs

```diff
@@ -261,11 +261,20 @@ public void Start()
 
         private void OnProcessMonitorTimer(object state)
         {
-            _process.Refresh();
-            var commitSizeBytes = _process.WorkingSet64;
-            if (commitSizeBytes != 0)
+            try
+            {
+                _process.Refresh();
+                var commitSizeBytes = _process.WorkingSet64;
+                if (commitSizeBytes != 0)
+                {
+                    AddMemoryActivity(DateTime.UtcNow, commitSizeBytes);
+                }
+            }
+            catch (Exception e)
             {
-                AddMemoryActivity(DateTime.UtcNow, commitSizeBytes);
+                // throwing this exception will mask other underlying exceptions.
```

Comment:
Not sure we want to do this? This effectively makes memory monitoring "best effort". If we can't monitor, I don't think we want to run. What exceptions are you seeing from this code?

Reply:
So far i have seen System.InvalidOperationException: Process has exited, so the requested information is not available.
We could scope it to just this exception but not confident enough to say this is the only exception.

We could start throwing the exception here once we have health checks implemented. Currently throwing this will just leave the app in a permanently broken state.

Opened an issue to track:
https://github.com/Azure/azure-functions-host/issues/5736